app/src/main/java/com/cookpad/android/dokumiTestAndroid/MainActivity.java

```diff
@@ -36,4 +46,30 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
         return super.onOptionsItemSelected(item);
     }
+
+    private void nullDereference() {
+        Object foo = getValueRandomly();
+        Log.e(TAG, foo.toString());
+    }
+
+    private Object getValueRandomly() {
+        // return value could be null
+        if (Math.random() < 0.1) {
+            return null;
+        } else {
+            return new Object();
+        }
+    }
+
+    private void resourceLeak() {
+        byte[] arr = {1, 2, 3};
+        FileOutputStream fis;
+        try {
+            fis = new FileOutputStream("file.txt");
+            fis.write(arr);
+            fis.close();
+        } catch (IOException e) {
```

Comment:
**Static Analysis:**  resource acquired to fis by call to FileOutputStream(...) at line 68 is not released after line 71